src/expansion.rs

```diff
@@ -1,25 +1,32 @@
 use super::peg::Job;
-use super::Variables;
+use super::Shell;
 
-pub fn expand_variables(jobs: &mut [Job], variables: &Variables) {
-    for mut job in &mut jobs[..] {
-        job.command = expand_string(&job.command, variables).to_string();
-        job.args = job.args
-                      .iter()
-                      .map(|original: &String| expand_string(&original, variables).to_string())
-                      .collect();
-    }
+pub trait Expand {
```

Comment:
I need to ask @skylerberg if he would be okay to include them into `Shell` or if wants to keep the expansion stuff separate. I just made it into a trait to keep it modular for now. As for documentation, I don't know what say.